### PR TITLE
tidy: release add Team Editor feature

### PR DESCRIPTION
**Pull request by jamdelion**
Ticket: https://trello.com/c/piMcGsJa/2973-add-user-in-editor

## ✅ Done
- Add accessibility test
- Add error toast if server-side error (probably won't get reached?)
- Release from feature flag

### CODE REVIEW

File: editor.planx.uk/src/pages/FlowEditor/components/Team/components/AddNewEditorModal.tsx [hunk 1]
```diff
@@ -24,13 +24,15 @@ import { optimisticallyUpdateMembersTable } from "./lib/optimisticallyUpdateMemb
 export const AddNewEditorModal = ({
   showModal,
   setShowModal,
-  setShowToast,
+  setShowSuccessToast,
+  setShowErrorToast,
```
Comment (jamdelion) [inline editor.planx.uk/src/pages/FlowEditor/components/Team/components/AddNewEditorModal.tsx:28]: This is the sort of thing I hope to get rid of with a `useToast` hook in another PR 😁

File: editor.planx.uk/src/pages/FlowEditor/components/Team/components/AddNewEditorModal.tsx [hunk 2]
```diff
@@ -49,16 +51,19 @@ export const AddNewEditorModal = ({
       if (isUserAlreadyExistsError(err.message)) {
         setShowUserAlreadyExistsError(true);
       }
+      if (err.message === "Unable to create user") {
+        setShowErrorToast(true);
+      }
```
Comment (jamdelion) [inline editor.planx.uk/src/pages/FlowEditor/components/Team/components/AddNewEditorModal.tsx:56]: Not sure if we'll ever reach this due to the error handling already in `src/lib/graphql.ts` but hopefully a good fallback

File: editor.planx.uk/src/pages/FlowEditor/components/Team/components/MembersTable.tsx [hunk 3]
```diff
@@ -20,17 +19,29 @@ export const MembersTable = ({
...
+
+  const handleCloseErrorToast = (
+    _event?: React.SyntheticEvent | Event,
+    reason?: string,
+  ) => {
+    if (reason === "clickaway") {
+      return;
+    }
+
+    setShowErrorToast(false);
```
Comment (jamdelion) [inline editor.planx.uk/src/pages/FlowEditor/components/Team/components/MembersTable.tsx:44]: Leaving this as duplicated for now and will revisit with a useToast hook 🤞

File: editor.planx.uk/src/pages/FlowEditor/components/Team/queries/createAndAddUserToTeam.tsx [hunk 4]
```diff
@@ -11,7 +15,7 @@ export const createAndAddUserToTeam = async (
   teamSlug: string,
 ) => {
   // NB: the user is hard-coded with the 'teamEditor' role for now
-  const response = (await client.mutate({
+  const response: CreateAndAddUserResponse = await client.mutate({
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Team/queries/createAndAddUserToTeam.tsx:18]: FYI: `client.mutate` also accepts a generic which will automatically wrap the provided type with `FetchResult`, e.g. `client.mutate<MyType>`

**Review by DafyddLlyr — APPROVED** [review]
Great! All looks good,  just the small issue highlighted by @jessicamcinchak on Slack but then I think that's it 🙌

**Pull request merged**